ops/main.py

```diff
@@ -42,6 +43,38 @@
 logger = logging.getLogger()
 
 
+@dataclasses.dataclass(frozen=True)
+class _JujuContext:
+    # source: https://juju.is/docs/juju/charm-environment-variables
+    # the HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398
+    # only allow accesing these vars because these are the currently used vars in main.py
+
+    juju_relation: str = ''
+    juju_relation_id: str = ''
+    juju_secret_revision: str = ''
+    juju_workload_name: str = ''
```

Comment:
Why are these `str = ''` and the others `Optional[str] = None`? I would think we should use `Optional[str] = none` for all.